addons/im_livechat/static/src/models/thread/thread.js

```diff
@@ -39,23 +39,55 @@ registerClassPatchModel('mail.thread', 'im_livechat/static/src/models/thread/thr
                  * of polluting the database, it is therefore acceptable and
                  * easier to handle one temporary partner per channel.
                  */
-                data2.members.push(unlink(this.env.messaging.publicPartners));
                 const partner = this.env.models['mail.partner'].create(
                     Object.assign(
                         this.env.models['mail.partner'].convertData(data.livechat_visitor),
                         { id: this.env.models['mail.partner'].getNextPublicId() }
                     )
                 );
-                data2.members.push(link(partner));
-                data2.correspondent = link(partner);
+                const publicPartnerIds = new Set(this.env.messaging.publicPartners.map(partner => partner.id));
+                this._setPartnerCommands({
+                    data: data2,
+                    predicate: (id) => publicPartnerIds.has(id),
+                    command: link(partner),
```

Comment:
This can/should be `insert` too actually. `convertData` is supposed to be without side effect, and the previous code was creating the fake partner inside it (line 42, this should be changed). Instead it should be created/inserted through the resulting command here.